test/test_jit_py3.py

```diff
@@ -678,6 +678,56 @@ def attr(self):
         with self.assertRaisesRegex(torch.nn.modules.module.ModuleAttributeError, "has no attribute"):
             scripted_mod.ignored_attr
 
+    def test_ignoring_module_attributes(self):
```

Comment:
1. can we add tests to `test_type_sharing.py` to make sure we are not improperly sharing types with different ignored attributes sets?
2. I think we should stop adding to `test_jit_py3.py`. This file originally existed because it let us use py3 syntax, but that no longer has a purpose now that we are py3 only.

Reply:
1. I added tests to `test_type_sharing.py` to test that types are shared if ignoring attrs make them equal and not shared if the ignored attributes list is modified and the types of modules before and after the modification should be different.
2. Where should I move them?

Reply:
For 2: not sure, I wish we had the tests that exercise "compilation customization" APIs, but they have not yet been factored out of `test_jit.py`.